quine-endpoints/src/main/scala/com/thatdot/quine/routes/IngestRoutes.scala

```diff
@@ -145,7 +145,24 @@ sealed abstract class KafkaSecurityProtocol(val name: String)
 object KafkaSecurityProtocol {
   case object PlainText extends KafkaSecurityProtocol("PLAINTEXT")
   case object Ssl extends KafkaSecurityProtocol("SSL")
-  val values: Seq[KafkaSecurityProtocol] = Seq(PlainText, Ssl)
+  case object Sasl_Ssl extends KafkaSecurityProtocol("SASL_SSL")
+  val values: Seq[KafkaSecurityProtocol] = Seq(PlainText, Ssl, Sasl_Ssl)
+}
+
+@title("Kafka SASL Authentication")
+@docs(
+  "See [SASL authentication in the Kafka documentation](https://docs.confluent.io/4.1.3/kafka/authentication_sasl.html)."
+)
+sealed abstract class KafkaSaslAuthentication
+object KafkaSaslAuthentication {
+  @title("Kafka SASL PLAIN Authentication")
+  @docs(
+    "See [SASL PLAIN authentication in the Kafka documentation](https://docs.confluent.io/4.1.3/kafka/authentication_sasl.html#plain)."
+  )
+  final case class Plain(
+    jaasConfig: String = "org.apache.kafka.common.security.plain.PlainLoginModule required username='' password='';",
```

Comment:
I think this class could just take the username and password fields, and we could construct this string from those, rather than making the end-user supply the entire JAAS config?